test/integration/buildrun_cleanup_test.go

```diff
@@ -0,0 +1,293 @@
+		Expect(err).To(BeNil())
+	})
+
+	AfterEach(func() {
+		_, err = tb.GetBuild(buildObject.Name)
+		if err == nil {
+			Expect(tb.DeleteBuild(buildObject.Name)).To(BeNil())
+		}
+
+		err := tb.DeleteClusterBuildStrategy(cbsObject.Name)
+		Expect(err).To(BeNil())
+	})
+
+	JustBeforeEach(func() {
+		if buildSample != nil {
+			buildObject, err = tb.Catalog.LoadBuildWithNameAndStrategy(BUILD+tb.Namespace, STRATEGY+tb.Namespace, buildSample)
+			Expect(err).To(BeNil())
+		}
+
+		if buildRunSample != nil {
+			buildRunObject, err = tb.Catalog.LoadBRWithNameAndRef(BUILDRUN+tb.Namespace, BUILD+tb.Namespace, buildRunSample)
+			Expect(err).To(BeNil())
+		}
+	})
+
+	Context("When a buildrun related to a build with short ttl set succeeds", func() {
+
+		BeforeEach(func() {
+			buildSample = []byte(test.MinimalBuildWithRetentionTTLFive)
+			buildRunSample = []byte(test.MinimalBuildRunRetention)
+		})
+
+		It("Should not find the buildrun after few seconds after it succeeds", func() {
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+			Expect(err).To(BeNil())
+
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+
+			br, err := tb.GetBRTillCompletion(buildRunObject.Name)
+			Expect(err).To(BeNil())
+			Expect(br.Status.GetCondition(v1alpha1.Succeeded).Status).To(Equal(corev1.ConditionTrue))
+			_, err = tb.GetBRTillNotFound(buildRunObject.Name, time.Second*1, time.Minute)
+			Expect(apierrors.IsNotFound(err)).To(BeTrue())
+		})
+	})
+
+	Context("Multiple successful buildruns related to build with limit 1", func() {
+
+		BeforeEach(func() {
+			buildSample = []byte(test.MinimalBuildWithRetentionLimitOne)
+			buildRunSample = []byte(test.MinimalBuildRunRetention)
+		})
+
+		It("The older successful buildrun should not exist", func() {
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+			Expect(err).To(BeNil())
+
+			// Create first buildrun
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-1"
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+			br1, err := tb.GetBRTillCompletion(buildRunObject.Name)
+			Expect(err).To(BeNil())
+			Expect(br1.Status.GetCondition(v1alpha1.Succeeded).Status).To(Equal(corev1.ConditionTrue))
+			// Create second buildrun
+			buildRunObject.Name = BUILDRUN + tb.Namespace + "-2"
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+			br2, err := tb.GetBRTillCompletion(buildRunObject.Name)
+			Expect(err).To(BeNil())
+			Expect(br2.Status.GetCondition(v1alpha1.Succeeded).Status).To(Equal(corev1.ConditionTrue))
+
+			_, err = tb.GetBR(BUILDRUN + tb.Namespace + "-1")
+			Expect(apierrors.IsNotFound(err)).To(BeTrue())
+		})
+
+	})
+
+	Context("Multiple buildruns with different build limits for failure and success", func() {
+
+		BeforeEach(func() {
+			cbsObjectFail, err = tb.Catalog.LoadCBSWithName(STRATEGY+tb.Namespace+"-fail", []byte(test.ClusterBuildStrategySingleStepNoPush))
```

Comment:
can we use the noop strategy? if yes, pls delete the definition of this clusterbuildstrategy.

Reply:
you can control the behaviour of the strategy, e.g. if u need an exit code 1 or 0. The control can be done with parameters or env vars in the buildrun.